contracts/TxPermission.sol

```diff
@@ -79,26 +78,34 @@ contract TxPermission is OwnedEternalStorage {
     function allowedTxTypes(
         address _sender,
         address _to,
-        uint256 /*_value*/, // solhint-disable-line space-after-comma
+        uint256 _value,
         uint256 _gasPrice
     )
         public
         view
         returns(uint32, bool)
     {
         if (_gasPrice > 0 || isSenderAllowed(_sender)) {
-            // Let `_sender` create any transactions with non-zero gas price
-            // or if he is in the allowedSenders list
+            // Let `_sender` create any transactions if any of the following are true:
+            // * the gas price is non-zero
+            // * they are in the allowedSenders list
+            // * they are the owner
             return (ALL, false);
         }
 
-        if (_to == RANDOM_CONTRACT && IValidatorSet(VALIDATOR_SET_CONTRACT).isValidator(_sender)) {
-            // Let the validator call any function of `Random` contract with zero gas price
+        if (0 != _value) {
+            // No other transactions are allowed to send Ether
+            return (NONE, false);
+        }
+
+        if (_to == RANDOM_CONTRACT && VALIDATOR_SET_CONTRACT.isValidator(_sender)) {
+            // Let the validator call any function of `Random` contract with zero gas price and zero ether
```

Comment:
Let's consistently capitalize "Ether"; or call it "native tokens"? Not sure.

Reply:
I agree: it's better to call it `native coins`.